lib/ansible/modules/clustering/kubevirt/kubevirt_vm.py

```diff
@@ -0,0 +1,312 @@
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_disk
+        type: list
+    volumes:
+        description:
+            - Volume specification for the virtual machine.
+            - https://kubevirt.io/api-reference/master/definitions.html#_v1_volume
+        type: str
+    persistent:
+        description:
+            - If I(true) and C(state) is I(running) virtual machine will be created and persisted, if I(false) and C(state) running non perisistent virtual machine will be created. 
+        type: bool
+
+extends_documentation_fragment:
+  - k8s_auth_options
+  - k8s_resource_options
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+'''
+
+EXAMPLES = '''
+- name: Start virtual machine 'myvm'
+  kubevirt_vm:
+      state: running
+      name: myvm
+      namespace: vms
+
+- name: Create virtual machine 'myvm' and start it
+  kubevirt_vm:
+      state: running
+      name: myvm
+      namespace: vms
+      memory: 64M
+      disks:
+        - name: registrydisk
+          volumeName: registryvolume
+          disk:
+            bus: virtio
+
+- name: Start non-persistent virtual machine 'myvm'
+  kubevirt_vm:
+      persistent: false
+      state: running
+      name: myvm
+      namespace: vms
+      memory: 64M
+      disks:
+        - name: registrydisk
+          volumeName: registryvolume
+          disk:
+            bus: virtio
+
+- name: Remove virtual machine 'myvm'
+  kubevirt_vm:
+      state: absent
+      name: myvm 
+      namespace: vms
+'''
+
+RETURN = '''
+vm:
+    description:
+      - The virtual machine that user manages.
+      - It contains all attributes https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachine
+    returned: success
+    type: complex
+'''
+
+import copy
+import traceback
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+from openshift.helper.exceptions import KubernetesException
+
+from openshift import watch
+from openshift.dynamic.client import ResourceInstance
+from openshift.helper.exceptions import KubernetesException
+
+
+VM_ARG_SPEC = {
+    
+    'merge_type': {'type': 'list', 'choices': ['json', 'merge', 'strategic-merge']},
+    'persistent': {'type': 'bool', 'default': True},
+    'state': {
+        'type': 'str',
+        'choices': [
+            'present', 'absent', 'running', 'stopped'
+        ],
+        'default': 'present'
+    },
+    'wait': {'type': 'bool', 'default': True},
+    'wait_time': {'type': 'int', 'default': 20},
+    'memory': {'type': 'str'},
+    'disks': {'type': 'list'},
+    'volumes': {'type': 'list'},
+}
+
+
+class KubeVirtVM(KubernetesRawModule):
+    def __init__(self, *args, **kwargs):
+        super(KubeVirtVM, self).__init__(*args, **kwargs)
+
+    @property
+    def argspec(self):
+        """ argspec property builder """
```

Comment:
I would like to keep the old code so we can still support `inline`. In this case it would be great to share the code.

Reply:
Yes it will be still supported.